t/plugin/elasticsearch-logger.t

```diff
@@ -0,0 +1,182 @@
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 2: Basic configuration verification
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.elasticsearch-logger")
+            local ok, err = plugin.check_schema({user = "elastic",
+                                                 password = "elastic",
+                                                 index = "apisix-elasticsearch-log",
+                                                 endpoint_addr = "http://127.0.0.1:10420/elasticsearch-logger/test"
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: auth configure undefined
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.elasticsearch-logger")
+            local ok, err = plugin.check_schema({user = "elastic",
+                                                 password = "elastic",
+                                                 index = "apisix-elasticsearch-log",
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+property "endpoint_addr" is required
+
+
+
+=== TEST 4: add plugin on routes
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "elasticsearch-logger": {
+                                "user": "elastic",
+                                "password": "elastic",
+                                "index": "apisix-elasticsearch-log",
+                                "endpoint_addr": "http://127.0.0.1:10420/elasticsearch-logger/test",
+                                "batch_max_size":1,
+                                "inactive_timeout":1
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:5601": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/opentracing"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 5: access local server
+--- request
+GET /opentracing
+--- response_body
+opentracing
+--- error_log
+elasticsearch body:
+elasticsearch headers: authorization:Basic ZWxhc3RpYzplbGFzdGlj
+--- wait: 5
```

Comment:
Not required I presume!

Reply:
for creating indices [each day] for storing logs，refer to the beats component's log storage ,i think it works well with the data time range

Reply:
> for creating indices [each day] for storing logs，refer to the beats component's log storage ,i think it works well with the data time range

@bisakhmondal

Reply:
The index in elasticsearch is unlike the one in RDMS. It is more like a partition or a log file.

AFAIK, the admin of elasticsearch creates a template to define the pattern of indexes, and the client can create any index with different suffix (usually a timestamp).

Elasticsearch will query each index parallelly. And log in old indexes can be purged to save the disk.

Reply:
> The index in elasticsearch is unlike the one in RDMS. It is more like a partition or a log file.
> 
> AFAIK, the admin of elasticsearch creates a template to define the pattern of indexes, and the client can create any index with different suffix (usually a timestamp).
> 
> Elasticsearch will query each index parallelly. And log in old indexes can be purged to save the disk.

You may be referring to the index lifecycle policies..

Reply:
Yes

Reply:
Yeah, Ack!